api/department/service.py

```diff
@@ -9,11 +9,11 @@ def get_all_departments(page, page_size, order_by, term_id=None):
         ids_subquery = Course.query.with_entities(Course.department_id).filter_by(term_id=term_id).distinct()
         query = query.filter(Department.id.in_(ids_subquery))
     sql_results = sort_and_paginate(query, order_by, page, page_size).all()
-    return [dep.as_dict() for dep in sql_results]
+    return (dep.as_dict() for dep in sql_results)
```

Comment:
https://stackoverflow.com/questions/43808180/python-3-x-list-comprehension-vs-tuple-generator
Switch back to list comprehension

Reply:
Why?